lib/discourse_canned_replies/topic_query_extension.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module DiscourseCannedReplies::TopicQueryExtension
+  def list_canned_replies
+    if SiteSetting.canned_replies_category.blank?
+      raise Discourse::SiteSettingMissing.new('canned_replies_category')
+    end
+
+    create_list(
+      :canned_replies,
+      { category: SiteSetting.canned_replies_category.to_i, per_page: 999_999 }
```

Comment:
I've created a hidden setting in e37f7cb977a4b9a396cddcbe39d09224246ec57f to configure this limit with a default value of 1000. It should be enough to cover 99.99% of use cases and it is still a number "small" enough to prevent things from blowing up if someone messes up in Settings.

In case the user really needs more it is a setting that can be configured in the console without changing the plugin

Reply:
Do you think it is Ok or 1000 is too much?